src/modelAnalysis/subspaces/checkScaling.m

```diff
@@ -0,0 +1,308 @@
+
+    vars = {'estLevel', 'scltol', 'matrixAS', 'nMets', 'nRxns', 'minS', 'maxS', ...
+            'rmin', 'imin', 'rmax', 'imax', 'cmin', 'jmin', 'cmax', 'jmax'};
+
+    % fill the structure with the properties
+    for i = 1:length(vars)
+        scalingProperties.(vars{i}) = eval(vars{i});
+    end
+
+    % print out a summary report
+    if printLevel > 0
+        fprintf('\n ------------------------ Scaling summary report ------------------------\n\n');
+        if isfield(model, 'description')
+            fprintf(' Name of model:                                %s\n', model.description);
+        end
+        fprintf(' Estimation level:                             %s (scltol = %1.2f)\n', estLevel, scltol);
+        fprintf(' Name of matrix:                               %s\n', matrixAS);
+        fprintf(' Size of matrix:\n');
+        fprintf('        * metabolites:                         %d\n', nMets);
+        fprintf('        * reactions:                           %d\n', nRxns);
+        fprintf(' Stoichiometric coefficients:\n');
+        fprintf('        * Minimum:                             %1.2f\n', minS);
+        fprintf('        * Maximum:                             %1.2f\n', maxS);
+    end
+
+    if isfield(model, 'lb')
+        scalingProperties.minLB = minLB;
+        scalingProperties.maxLB = maxLB;
+
+        if printLevel > 0
+            fprintf(' Lower bound coefficients:\n');
+            fprintf('        * Minimum:                             %1.2f\n', minLB);
+            fprintf('        * Maximum:                             %1.2f\n', maxLB);
+        end
+    end
+
+    if isfield(model, 'ub')
+        scalingProperties.minUB = minUB;
+        scalingProperties.maxUB = maxUB;
+
+        if printLevel > 0
+            fprintf(' Upper bound coefficients:\n');
+            fprintf('        * Minimum:                             %1.2f\n', minUB);
+            fprintf('        * Maximum:                             %1.2f\n', maxUB);
+        end
+    end
+
+    if printLevel > 0
+        fprintf(' Row scaling coefficients:\n');
+        fprintf('        * Minimum:                             %1.2e (row #: %d)\n', rmin, imin);
+        fprintf('        * Maximum:                             %1.2e (row #: %d)\n', rmax, imax);
+        fprintf(' Column scaling coefficients:\n');
+        fprintf('        * Minimum:                             %1.2e (column #: %d)\n', cmin, jmin);
+        fprintf('        * Maximum:                             %1.2e (column #: %d)\n\n', cmax, jmax);
+        fprintf(' ---------------------------------- Ratios --------------------------------\n\n');
+    end
+
+    if abs(minS) > 0 && abs(ratioS) > 0
+        scalingProperties.ratioS = ratioS;
+        if printLevel > 0
+            fprintf(' Ratio of stoichiometric coefficients:         %1.2e\n', ratioS);
+        end
+    end
+    if abs(ratioS) > 0
+        scalingProperties.ratioS_orderOfMag = abs(floor(log10(abs(ratioS))));
```

Comment:
If I understand correctly, instead of:
````Matlab
minS = full(min(min(S)));
maxS = full(max(max(S)));
if abs(minS) > 0
    ratioS = maxS / minS;
else
    ratioS = maxS;
end
````
it would be better to use the absolute values of the stoichiometric coefficients of `S`:
````Matlab
minS = full(min(min(abs(S))));
maxS = full(max(max(abs(S))));
if abs(minS) > 0
    ratioS = maxS / minS;
else
    ratioS = maxS;
end
````